### PR TITLE
Workflow: update, multiple smaller fixes

### PR DESCRIPTION
**Pull request by 0xc0170**
Update the workflow.  There were couple of changes recently.

- mergify addition
- multiple labels were missing in the docs
- add notes about internal things like git2jira or reviewers - people should be aware they are internal (we are asked what it means)
- PR template was changes

cc @ARMmbed/mbed-os-maintainers Please review

### CODE REVIEW

**Comment by 0xc0170** [comment]
Some labels are not yet in here, questionable:

- PM_accepted/declined
- Priority: High/Medium/Low
- parked
- Found in OOB
- mirror internally
- open for community contributions

Time to review these as well.

**Comment by 0xc0170** [comment]
I would remove these labels, not covering here:

PM_accepted/declined
parked
mirror internally

File: docs/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -37,7 +37,7 @@ You can submit Mbed OS bugs directly on [GitHub](https://github.com/ARMmbed/mbed
 
 The bug report should be reproducible (fails for others) and specific (where and how it fails). We will close insufficient bug reports.
 
-We copy issues reported on GitHub to our internal tracker and regularly triage them.
+We copy issues reported on GitHub to our internal tracker and regularly triage them. Our ciarcom bot adds internal tracking reference to each issue: "Internal Jira reference: https://jira.arm.com/browse/MBOTRIAGE-xxxx".
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md:41]: Add provides labels showing the current state of the mirrored issues.

File: docs/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -206,10 +197,10 @@ If a pull request is idle for more than two weeks, it will be closed. The author
...
 
-GitHub dismisses a reviewer's status after any change to the pull request commit history (such as adding a new commit or rebasing). Smaller changes, such as documentation edits or rebases on top of latest master, only require additional review by maintainers. Their approval is sufficient because a team assigned as a reviewer already approved the pull request.
+Mergify dismisses a reviewer's status after any change to the pull request commit history (such as adding a new commit or rebasing). Smaller changes, such as documentation edits or rebases on top of latest master, only require additional review by maintainers. Their approval is sufficient because a team assigned as a reviewer already approved the pull request.
 
 - Label: `needs: review`.
-- Time: Three days for reviewers to leave feedback after the maintainers add the label.
+- Time: Three days for reviewers to leave feedback after the autoreviewer bot add the label.
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md]: after the autoreviewer bot has added the label.

File: docs/contributing/guidelines/workflow.md [hunk 3]
```diff
@@ -227,10 +218,12 @@ A pull request in the "work needed" state requires additional work due to failed
 
 ### Ready for integration
 
-Maintainers merge pull requests during the internal gatekeeping meetings that occur three times a week. They can merge straightforward pull requests immediately.
+A pull request is ready for merge when the pull request completed the reviews and tests.
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md]: A pull request is ready for merge once all the reviews and tests are complete.

File: docs/contributing/guidelines/workflow.md [hunk 4]
```diff
@@ -227,10 +218,12 @@ A pull request in the "work needed" state requires additional work due to failed
...
+
+Maintainers merge pull requests as they have write access to the main master branch. Pull request integration happens at any time of the day.
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md:235]: This may be automated in the future.

File: docs/contributing/guidelines/workflow.md [hunk 5]
```diff
@@ -254,16 +247,26 @@ alpha/beta release etc
 We use many other labels to summarize the scope and effect of the changes:
 
 - *needs: preceding PR* - Cannot yet be merged because it has a dependency on another pull request that needs to merge first.
-- *DO NOT MERGE* - Contains changes that may be in a draft state and submitted purely for review, or may contain breaking changes that have not been considered.
+- *do not merge* - Contains changes that may be in a draft state and submitted purely for review, or may contain breaking changes that have not been considered.
 - *devices: 'name'* - Specifically affects the named device(s).
 - *component: 'name'* - Specifically affects the named component. Component names follow the structure of Mbed OS (for example `ble`, `storage`, `tls`).
 - *rollup PR* - Ready for CI testing, but will instead be brought into a rollup pull request to test multiple pull requests at once.
+- *Release review required* - additional release version approval is required
+- *release version missing* - a merged pull request does not contain release label, it should be fixed by the maintainers team
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md]: does not contain a release label

File: docs/contributing/guidelines/workflow.md [hunk 6]
```diff
@@ -254,16 +247,26 @@ alpha/beta release etc
...
+- *BREAKING-CHANGE* - a pull request introducing breaking change
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md]: - this pull request introduces a breaking change

File: docs/contributing/guidelines/workflow.md [hunk 7]
```diff
@@ -254,16 +247,26 @@ alpha/beta release etc
...
+- *mirrored* - the git2jira bot mirrored an issue to our internal Jira
+- *JIRA status:* - an internal jira ticket status propagated to Github issue. It can be: OPEN, IN PROGRESS, CLOSED, RESOLVED or REOPENED.
+- *closed in jira* - mirrored internal Jira issue was closed
+- *Manually patch* - a pull request needs manual cherry-picking for the release candidate to resolve conflicts
+- *rollup PR* - rollup pull request, more details in the section below
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md]: Do we even need this one any more? The roll ups were complex at best...?

File: docs/contributing/guidelines/workflow.md [hunk 8]
```diff
@@ -254,16 +247,26 @@ alpha/beta release etc
...
+- *risk: x* - How big risks is the pull request for a specific release. Options are: green, amber or red.
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md]: How big a risk is the pull request

File: docs/contributing/guidelines/workflow.md [hunk 9]
```diff
@@ -254,16 +247,26 @@ alpha/beta release etc
...
```
Comment (adbridge) [inline docs/contributing/guidelines/workflow.md:273]: Missing: Open for community contribution ?

**Review by adbridge — CHANGES_REQUESTED** [review]

File: docs/contributing/guidelines/workflow.md (hunk 7 above)
Comment (0xc0170) [reply docs/contributing/guidelines/workflow.md]: I think we can remove them, it is there entire section about them

File: docs/contributing/guidelines/workflow.md (hunk 9 above)
Comment (0xc0170) [reply docs/contributing/guidelines/workflow.md:273]: yes, will add. See https://github.com/ARMmbed/mbed-os-5-docs/pull/1222#issuecomment-582910892 . lets discuss these and I'll update the labels here and possibly remove some not used anymore

**Comment by adbridge** [comment]
> 
> 
> I would remove these labels, not covering here:
> 
> PM_accepted/declined
> parked
> mirror internally

Mirror internally is used by the bot. If a particular repo is set to manual mirroring rather than automatic mirroring then only those PRs that have this label applied will be mirrored. That should definitely be in our docs.

**Comment by adbridge** [comment]
> 
> 
> Some labels are not yet in here, questionable:
> 
>     * PM_accepted/declined
> 
>     * Priority: High/Medium/Low
> 
>     * parked
> 
>     * Found in OOB
> 
>     * mirror internally
> 
>     * open for community contributions
> 
> 
> Time to review these as well.

Are you proposing a separate PR for these ?

**Comment by 0xc0170** [comment]
@adbridge Fixed all comments, plus added community contribution label. What about the rest?

>Are you proposing a separate PR for these ?

I can clean them here as I am already adding/fixing labels.

**Comment by 0xc0170** [comment]
Lets remove labels in separate PR. I'll document `mirror internally`, won't touch any other

**Comment by 0xc0170** [comment]
Added mirror internally label

**Review by adbridge — APPROVED** [review]

**Comment by 0xc0170** [comment]
I removed duplicated roll-up (there's PR removing this).

**Review by adbridge — APPROVED** [review]

**Comment by 0xc0170** [comment]
@AnotherButler just checking as I had this opened today - when this can be integrated?

**Comment by AnotherButler** [comment]
@0xc0170 I plan to merge this later this week. Does it also apply to 5.15?

**Review by AnotherButler — APPROVED** [review]

**Comment by 0xc0170** [comment]
> @0xc0170 I plan to merge this later this week. Does it also apply to 5.15?

I recall replying here but can't find my answer. Yes it does apply. As this is about our workflow.

**Pull request merged**